### ISSUE TITLE
Use block body throws out whitespace

### ISSUE THREAD

**Issue by miloush** [dotnet/roslyn#21341]
**Version Used**: 2.3.2.61928

**Steps to Reproduce**:

```C#
        public static IWebHost BuildWebHost(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build();
```
Invoke 'Use block body for methods' refactoring.

**Actual Behavior**:
```C#
        public static IWebHost BuildWebHost(string[] args)
        {
            return WebHost.CreateDefaultBuilder(args)
.UseStartup<Startup>()
.Build();
        }
```

**Expected Behavior**:
```C#
        public static IWebHost BuildWebHost(string[] args)
        {
            return WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build();
        }
```
(preserve absolute whitespace) or
```C#
        public static IWebHost BuildWebHost(string[] args)
        {
            return WebHost.CreateDefaultBuilder(args)
                       .UseStartup<Startup>()
                       .Build();
        }
```
(preserve relative whitespace) or any other handling, but no one wants it to be removed completely.

**Comment by CyrusNajmabadi** [comment]
@heejaechang Do you know why this is happening?  We're not touching any trivia inside the expression.  It's unclear to me why the formatting is being removed.

**Comment by heejaechang** [comment]
@CyrusNajmabadi anchor rule (where to put .UserStartxxxx) is hard to work when elastic trivia is involved. I believe anchor rule is monitoring "public"'s delta to find out where it should move .UserStartxxx

does the refactoring touch method signature when converting => to block?

**Comment by CyrusNajmabadi** [comment]
> does the refactoring touch method signature

No.  All we do is:

```c#
                    return methodDeclaration.WithExpressionBody(null)
                                            .WithSemicolonToken(default)
                                            .WithBody(block)
                                            .WithAdditionalAnnotations(Formatter.Annotation);
```

**Comment by heejaechang** [comment]
ah. played a bit and I think I know what is going on. when => is used, those WebHost ... expression is set to follow "public", but when it is changed to "{  return }", it is set to follow "return". since those "return" involves elastic trivia, those expression lost where to put those expression. 

I think you can either filter out any anchor rule so that code don't move, or put leading trivia to "return" same leading trivia as "Webhost", so anchor rule still work.

**Comment by heejaechang** [comment]
I think my guess is somewhat right, if I do this.

public static IWebHost BuildWebHost(string[] args) => /*hello*/
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build();

it works fine. if I do this

public static IWebHost BuildWebHost(string[] args) => 
            /*hello*/ WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build()

things are broken.

basically, if you put right trivia before return, it will work.

**Comment by CyrusNajmabadi** [comment]
> basically, if you put right trivia before return, it will work.

Thanks!  I'll try that.

**Comment by CyrusNajmabadi** [comment]
Unfortunately, even adding the appropriate whitspace directly to the 'return' keyword did not help here :(

**Comment by zaytsev-victor** [comment]
@CyrusNajmabadi, I've found a way how to fix it. If we use ```editor.SetStatements(declaration, statements)``` instead of ```editor.ReplaceNode(declaration, updatedDeclaration)``` formatting will be correct. 

If you ok with this changes I can grab the bug.

**Comment by CyrusNajmabadi** [comment]
I'd like to know what that changes anything, as they should be doing effectively the same thing.

**Comment by CyrusNajmabadi** [comment]
I don't repro this.  We now generate:

![image](https://user-images.githubusercontent.com/4564579/198749600-fba816c7-49e4-4a2e-b945-92850c211d72.png)

Which, while not ideal, is acceptable as it is at least preserving the whitespace.
